Applications/Spire/Source/Ui/ListCurrentController.cpp

```diff
@@ -69,11 +69,14 @@ void ListCurrentController::move(int source, int destination) {
     return 1;
   }();
   if(direction == 1) {
-    std::rotate(m_views.begin() + source, m_views.begin() + source + 1,
-      m_views.begin() + destination + 1);
+    std::rotate(m_views.begin() + destination, m_views.begin() + source,
```

Comment:
What does this have to do with the task?

Reply:
When the regions inside the `RegionBox` are sorted,  the `ListItem` containing the region Tags will be moved, and this function is to move `ListItems` from the `source` position to the `destination` position. `std::rotate` was used in the wrong way, so I have to try to fix it.

> template< class ForwardIt >
> void rotate( ForwardIt first, ForwardIt n_first, ForwardIt last );
>
> It accepts three parameters: 
> first       -the beginning of the original range
> n_first   -the element that should appear at the beginning of the rotated range
> last        -the end of the original range

If `source` > `destination`, `first` is `destination`, `n_first` is `source` and `last` is `source + 1`, which means to move the value at the `source` position to the beginning of the range `[destination, source + 1)`;
if `source` < `destination`, it moves the value at the `source` position to the end of the range `(source -1, destination]`,  using a reverse iterator.

Reply:
There are similar snippets of moving items inside `ListView` (`ListView::move_item`, line 394 and line 405 in ListView.cpp). I will update the code here to be consistent with the snippets inside `ListView::move_item`.